File: backend/app/schemas/dispatch.py
from pydantic import BaseModel
from typing import Optional
from enum import Enum
from datetime import datetime

class DispatchStatusEnum(str, Enum):
    CREATED = "CREATED"
    ASSIGNED = "ASSIGNED"
    ACCEPTED = "ACCEPTED"
    EN_ROUTE = "EN_ROUTE"
    ARRIVED = "ARRIVED"
    COMPLETED = "COMPLETED"
    CANCELLED = "CANCELLED"

class DispatchCreate(BaseModel):
    emergencyId: str
    driverId: str
    hospitalId: str

class DispatchResponse(BaseModel):
    id: str
    emergencyId: str
    driverId: str
    hospitalId: str
    status: str
    assignedAt: Optional[datetime] = None
    acceptedAt: Optional[datetime] = None
    completedAt: Optional[datetime] = None
    createdAt: datetime
    updatedAt: datetime

    class Config:
        from_attributes = True

File: frontend/lib/features/dispatch/domain/entities/dispatch.dart
import 'dispatch_status.dart';

class Dispatch {
  final String id;
  final String emergencyId;
  final String driverId;
  final String hospitalId;
  final DispatchStatus status;
  final DateTime? assignedAt;
  final DateTime? acceptedAt;
  final DateTime? completedAt;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Dispatch({
    required this.id,
    required this.emergencyId,
    required this.driverId,
    required this.hospitalId,
    required this.status,
    this.assignedAt,
    this.acceptedAt,
    this.completedAt,
    required this.createdAt,
    required this.updatedAt,
  });

  Dispatch copyWith({
    String? id,
    String? emergencyId,
    String? driverId,
    String? hospitalId,
    DispatchStatus? status,
    DateTime? assignedAt,
    DateTime? acceptedAt,
    DateTime? completedAt,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Dispatch(
      id: id ?? this.id,
      emergencyId: emergencyId ?? this.emergencyId,
      driverId: driverId ?? this.driverId,
      hospitalId: hospitalId ?? this.hospitalId,
      status: status ?? this.status,
      assignedAt: assignedAt ?? this.assignedAt,
      acceptedAt: acceptedAt ?? this.acceptedAt,
      completedAt: completedAt ?? this.completedAt,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}
